### DISCUSSION TITLE
Back relation query bug when child has two parent tables

### DISCUSSION THREAD

**Discussion by calebfaruki** [pocketbase/pocketbase#4567] [Q&A] [answered]
pocketbase version 0.22.4

I have three collections: `products`, `brands`, and `categories`. Products belong to a category and a brand.

For a given category, I want to get all the brands for the set of products. This works **sometimes** but not always.

What is wrong with this query?

```
products_via_brand_id.category_id = '<CATEGORY-ID>'
```

<img width="460" alt="Screenshot 2024-03-15 at 00 16 58" src="https://github.com/pocketbase/pocketbase/assets/5914427/4f4eb569-a87a-47ed-bf89-9b6834220f1e">
<img width="228" alt="Screenshot 2024-03-15 at 00 16 25" src="https://github.com/pocketbase/pocketbase/assets/5914427/c1715149-8e58-4767-88b2-42db776c26f3">

As you can see, TRP and Hayes aren't being returned even though I have products assigned to that brand.

This doesn't happen all the time. Here is what I do normally (and there's no issue):

1. Create brand record.
2. Create product record assigned to brand record.

**EDIT**: It seems to have to do with when I create products in two different categories for the same brand...

**Comment by ganigeorgiev** [comment, marked as answer]
> It seems to have to do with when I create products in two different categories for the same brand...

Yes, this is because the back-relation reference is resolved as multiple relation field. It should work if you replace your operator with `?=`.

For this and other caveats, please check the ["Back-relation caveats" section](https://pocketbase.io/docs/working-with-relations/#back-relation-caveats).
